### DISCUSSION TITLE
Proposal for Enhanced Import Functionality

### DISCUSSION THREAD

**Discussion by abadfox233** [deepnoodle-ai/risor#111] [Ideas]
I hope this message finds you well. I'd like to propose a couple of enhancements to the import functionality within Risor, which I believe would greatly benefit its users.

* Support for Multi-Level Imports:
Currently, Risor supports basic imports, such as import module . However, adding support for multi-level imports, such as `from a.b.c import d`, would significantly enhance the flexibility and usability of the library. This capability will allow users to directly import specific components nested within modules, providing a more granular and efficient import process.

* Support for Import Alias/Name Change:
Another useful feature would be the ability to rename imported modules or functions during the import process. For instance, enabling syntax like `from a.b.c import d as e` would provide developers with the flexibility to use an alternative name for an imported component, enhancing code readability and reducing potential naming conflicts.

These enhancements align with modern Python coding practices and would contribute to making Risor more versatile and user-friendly for a broader range of use cases.

I'm eager to contribute to this implementation if the proposal aligns with the project's vision. Your feedback and thoughts on these suggested improvements would be greatly appreciated.

Thank you for considering these enhancements.

Best regards

**Comment by myzie** [comment]
Hi there @abadfox233. I'm definitely supportive of these suggestions, and agree they would be useful additions. I had considered work on this before, but just haven't gotten to it yet.

Please do contribute on the implementation and I'll be happy to help test and provide feedback along the way.

**Comment by myzie** [comment]
Thanks @abadfox233 for the contribution. I'm marking this discussion thread as closed.

The relevant PR: https://github.com/risor-io/risor/pull/116
